geo/src/main/java/org/odk/collect/geo/SelectionMapFragment.kt

```diff
@@ -321,41 +340,23 @@ class SelectionMapFragment : Fragment() {
     }
 }
 
-class SelectionMapViewModel : ViewModel() {
+class SelectionViewModel : ViewModel() {
 
-    private var mapTitle = MutableLiveData<String>()
-    private var mappableItems = MutableLiveData<List<MappableSelectItem>>(emptyList())
-    private var itemCount = MutableLiveData(0)
     private var selectedItemId: Int? = null
 
-    fun getMapTitle(): LiveData<String> {
-        return mapTitle
-    }
-
     fun getSelectedItemId(): Int? {
         return selectedItemId
     }
 
     fun setSelectedItemId(itemId: Int?) {
         selectedItemId = itemId
     }
+}
 
-    fun getItemCount(): LiveData<Int> {
-        return itemCount
-    }
-
-    fun getMappableItems(): LiveData<List<MappableSelectItem>> {
-        return mappableItems
-    }
-
-    fun setItems(itemCount: Int, mappableItems: List<MappableSelectItem>) {
-        this.mappableItems.value = mappableItems
-        this.itemCount.value = itemCount
-    }
-
-    fun setMapTitle(title: String) {
-        this.mapTitle.value = title
-    }
+abstract class SelectionMapViewModel : ViewModel() {
+    abstract fun getMapTitle(): LiveData<String>
+    abstract fun getItemCount(): LiveData<Int>
+    abstract fun getMappableItems(): NonNullLiveData<List<MappableSelectItem>>
 }
 
 data class MappableSelectItem(
```

Comment:
We'll probably need to make the `status` field here a list of `IconifiedText` so we can display arbitrary columns from GeoJSON, but it feels like we should wait to get there before doing that.